src/server.js

```diff
@@ -1,13 +1,70 @@
-import * as dotenv from "dotenv";
-dotenv.config();
-import express from "express";
-import allRoutes from "./routes/index";
+import * as dotenv from 'dotenv';
+import express from 'express';
+import session from 'express-session';
+import passport from 'passport';
+import jwt from 'jsonwebtoken';
+import BcryptUtil from 'bcrypt';
+import welcomeRoutes from './routes/welcomeRoute';
+import allRoutes from './routes/index';
+import './config/passport';
+import googleCallback from './middlewares/googleCallback';
+import db from './models/index';
+
+const User = db['User'];
 
 export const app = express();
 
+dotenv.config();
+
 app.use(express.json());
 app.use(express.urlencoded({ extended: true }));
 
-app.use("/", allRoutes);
+app.use('/', allRoutes);
+app.use(
+  session({
+    secret: process.env.EXPRESS_SESSION,
+    resave: false,
+    saveUninitialized: false,
+  }),
+);
+app.use('/welcome', welcomeRoutes);
+
+app.get('/', (req, res) => {
+  res.send('<a href="/auth/google">Authenticate with Google</a>');
+});
+app.get(
+  '/auth/google',
+  passport.authenticate('google', { scope: ['email', 'profile'] }),
+);
+
+app.get('/google/callback', googleCallback, async (req, res) => {
+  try {
+    const { given_name, email } = req.user;
+    const googleUser = await User.findOne({ where: { email } });
+    let userObject = {};
+    if (googleUser) {
+      const TOKEN = jwt.sign(
+        { name: given_name, email },
```

Comment:
create helper function that can generate token because you team they might need in  future